.github/workflows/build-image.yml

```diff
@@ -0,0 +1,23 @@
+name: Build image and publish to Docker Hub
+
+on:
+  push:
+    tags:
+      - '*'
```

Comment:
does this mean that it'll only build if we make a tag? What about making it a bit more specific? Can we ensure that the tag is formatted correctly?

Reply:
Yes, it'll only build and push the image on a tag. Yes, I think it supports regex so we can restrict the build to properly formatted tags.

Reply:
We can do something like `v-[0-9]{4}-[0-9]{2}-[0-9]{1,2}.[0-9]+` instead of `*` to trigger the workflow only if the tag matches the regex.